microovn/ovn/services.go

```diff
@@ -0,0 +1,69 @@
+package ovn
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"github.com/canonical/microcluster/state"
+	"github.com/canonical/microovn/microovn/database"
+)
+
+func DisableService(s *state.State, service string) error {
+	err := s.Database.Transaction(s.Context, func(ctx context.Context, tx *sql.Tx) error {
+		exists, err := database.ServiceExists(ctx, tx, s.Name(), service)
+		if err != nil {
+			return err
+		}
+		if exists != true {
+			return errors.New("No such service")
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	err = snapStop(service, true)
```

Comment:
Some of our services, such as `chassis` and `switch` are simple and map directly to a single systemd service.

However we do also have more complex services, such as `central` which in reality map into multiple systemd services:
https://github.com/canonical/microovn/blob/a99570ddbcffe6acbd76698a424c78d3cca49773/microovn/ovn/join.go#L113-L126

This is a useful abstraction and we want to keep the components individually controllable at the lower level.

However, for the purpose of the `microovn enable` and `microovn disable` commands we need some sort of abstraction so that the user can do `microovn disable central` and have the system disable all the underlying systemd services that compose that higher level service.